src/tribol/physics/CommonPlane.cpp

```diff
@@ -247,22 +241,24 @@ int ApplyNormal< COMMON_PLANE, PENALTY >( CouplingScheme const * cs )
       {
          case KINEMATIC_CONSTANT: 
          {
-            // Average each mesh's penalty stiffness premultiplied by each mesh's penalty scale
-            penalty_stiff_per_area  = 0.5 * 
-                                      ( pen_scale1 * mesh1.m_elemData.m_penalty_stiffness +
-                                        pen_scale2 * mesh2.m_elemData.m_penalty_stiffness );
+            // pre-multiply each spring stiffness by each mesh's penalty scale
+            auto stiffness1 = pen_scale1 * mesh1.m_elemData.m_penalty_stiffness;
+            auto stiffness2 = pen_scale2 * mesh2.m_elemData.m_penalty_stiffness;
+            // compute the equivalent contact penalty spring stiffness per area
+            penalty_stiff_per_area  = ComputePenaltyStiffnessPerArea( stiffness1, stiffness2 );
             break;
          }
          case KINEMATIC_ELEMENT:
          {
-            // multiply the material modulus (i.e. material stiffness) by each mesh's penalty scale
-            penalty_stiff_per_area = ComputePenaltyStiffnessPerArea( 
-                                        pen_scale1 * mesh1.m_elemData.m_mat_mod[ index1 ],
-                                        mesh1.m_elemData.m_thickness[ index1 ],
-                                        pen_scale2 * mesh2.m_elemData.m_mat_mod[ index2 ],
-                                        mesh2.m_elemData.m_thickness[ index2 ],
-                                        pen_enfrc_options.tiny_length
-                                                                   );
+            // add tiny_length to element thickness to avoid division by zero
+            auto t1 = mesh1.m_elemData.m_thickness[ index1 ] + pen_enfrc_options.tiny_length;
```

Comment:
Not sure if it's checked elsewhere, but would it be worth it to add a check for negative thicknesses?

Reply:
I added a check to ApplyNormal<>. If any face has a negative thickness the routine will return an error at the end of the physics loop after printing a debug message.